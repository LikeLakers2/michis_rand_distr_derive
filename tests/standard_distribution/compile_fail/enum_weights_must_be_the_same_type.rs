use michis_rand_distr_derive::StandardDistribution;
fn main() {}

#[derive(StandardDistribution)]
pub enum ErrorTest1 {
	#[standard_distribution(weight = 1.0f32)]
	Variant1,
	#[standard_distribution(weight = 3.0f64)]
	Variant2,
}

#[derive(StandardDistribution)]
pub enum ErrorTest2 {
	#[standard_distribution(weight = 1.0)]
	Variant1,
	#[standard_distribution(weight = 3)]
	Variant2,
}
