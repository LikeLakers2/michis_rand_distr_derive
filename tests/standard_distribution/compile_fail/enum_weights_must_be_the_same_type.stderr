error[E0308]: mismatched types
 --> tests/standard_distribution/compile_fail/enum_weights_must_be_the_same_type.rs:8:35
  |
8 |     #[standard_distribution(weight = 3.0f64)]
  |                                      ^^^^^^ expected `f32`, found `f64`
  |
help: change the type of the numeric literal from `f64` to `f32`
  |
8 |     #[standard_distribution(weight = 3.0f32)]
  |                                         ~~~

error[E0308]: mismatched types
  --> tests/standard_distribution/compile_fail/enum_weights_must_be_the_same_type.rs:16:35
   |
16 |     #[standard_distribution(weight = 3)]
   |                                      ^ expected floating-point number, found integer
